src/cluster/src/start/k8.rs

```diff
@@ -781,6 +791,36 @@ impl ClusterInstaller {
             }
         }
 
+        // When upgrading, wait for platform version to match new version
```

Comment:
This seems duplicate of sc checking? Can we consolidate to there.  Version check is useful as there too

Reply:
Correct me if I'm wrong but it seems like the upgrade process would not need `create_managed_spu_group`? Aside from that, I can see if I can consolidate

Reply:
Yes creating spus is not needed in upgrading.  It would be good if we can move post install/upgrade workflow as separate entity (with builder?) so it can be more flexible

Reply:
Yes, I have plans like that for when we refactor this crate